### PR TITLE
[FLINK-14395] Refactor ES 6 connector to split table-specific code into flink-sql-connector-elasticsearch6

### PR DESCRIPTION
**Pull request by yanghua**
## What is the purpose of the change

*This pull request try to refactor ES 6 connector to split table-specific code into flink-sql-connector-elasticsearch6*

## Brief change log

  - *Moved table-specific code from elasticsearch 6.x general connector to sql connector*
  - *changed pom file and some resource files*

## Verifying this change

This change is a trivial rework / code cleanup without any test coverage.

## Does this pull request potentially affect one of the following parts:

  - Dependencies (does it add or upgrade a dependency): (yes / **no**)
  - The public API, i.e., is any changed class annotated with `@Public(Evolving)`: (yes / **no**)
  - The serializers: (yes / **no** / don't know)
  - The runtime per-record code paths (performance sensitive): (yes / **no** / don't know)
  - Anything that affects deployment or recovery: JobManager (and its components), Checkpointing, Yarn/Mesos, ZooKeeper: (yes / **no** / don't know)
  - The S3 file system connector: (yes / **no** / don't know)

## Documentation

  - Does this pull request introduce a new feature? (yes / **no**)
  - If yes, how is the feature documented? (not applicable / docs / JavaDocs / **not documented**)

### CODE REVIEW

**Comment by yanghua** [comment]
I have reported the Kafka test issue on Travis. Trying to rebuild now.

**Comment by yanghua** [comment]
@flinkbot run travis

**Comment by yanghua** [comment]
@flinkbot run travis

**Comment by yanghua** [comment]
Still failed because of the same Kafka relevant test failure.

**Comment by yanghua** [comment]
@flinkbot run travis

File: flink-connectors/flink-sql-connector-elasticsearch6/src/test/java/org/apache/flink/streaming/sql/connectors/elasticsearch6/Elasticsearch6UpsertTableSinkFactoryTest.java [hunk 1]
```diff
@@ -65,11 +65,11 @@ public void testBuilder() {
 		final TestElasticsearch6UpsertTableSink testSink = new TestElasticsearch6UpsertTableSink(
 			false,
 			schema,
-			Collections.singletonList(new Host(HOSTNAME, PORT, SCHEMA)),
-			INDEX,
-			DOC_TYPE,
-			KEY_DELIMITER,
-			KEY_NULL_LITERAL,
+			Collections.singletonList(new Host(ElasticsearchUpsertTableSinkFactoryTestBase.HOSTNAME, ElasticsearchUpsertTableSinkFactoryTestBase.PORT, ElasticsearchUpsertTableSinkFactoryTestBase.SCHEMA)),
```
Comment (aljoscha) [inline flink-connectors/flink-sql-connector-elasticsearch6/src/test/java/org/apache/flink/streaming/sql/connectors/elasticsearch6/Elasticsearch6UpsertTableSinkFactoryTest.java]: Shouldn't this work without the `ElasticsearchUpsertTableSinkFactoryTestBase` prefix? We have `ElasticsearchUpsertTableSinkFactoryTestBase` as our base class.

File: flink-connectors/flink-sql-connector-elasticsearch6/pom.xml [hunk 2]
```diff
@@ -36,11 +36,65 @@ under the License.
...
+				</exclusion>
+			</exclusions>
+		</dependency>
+
 		<dependency>
 			<groupId>org.apache.flink</groupId>
 			<artifactId>flink-connector-elasticsearch6_${scala.binary.version}</artifactId>
 			<version>${project.version}</version>
 		</dependency>
+
```
Comment (aljoscha) [inline flink-connectors/flink-sql-connector-elasticsearch6/pom.xml]: These should now be removed from the pure ES6 connector because they are not needed anymore.

**Comment by yanghua** [comment]
> I don't think this will still work because this was not changed:
> 
> https://github.com/apache/flink/blob/10f9f1d431dbf047ea14130d9fb7fdd4f78178fc/flink-connectors/flink-connector-elasticsearch6/src/main/resources/META-INF/services/org.apache.flink.table.factories.TableFactory#L16
> 
> . And you changed the name of the package.

I have moved the file from `flink-connector-elasticsearch6` to `flink-sql-connector-elasticsearch6` just like what I did in the PR #9720 . So it should not be a problem?

File: flink-connectors/flink-sql-connector-elasticsearch6/src/test/java/org/apache/flink/streaming/sql/connectors/elasticsearch6/Elasticsearch6UpsertTableSinkFactoryTest.java (hunk 1 above)
Comment (yanghua) [reply flink-connectors/flink-sql-connector-elasticsearch6/src/test/java/org/apache/flink/streaming/sql/connectors/elasticsearch6/Elasticsearch6UpsertTableSinkFactoryTest.java]: You are right. The IDE added the prefix when I moving the class files.

File: flink-connectors/flink-sql-connector-elasticsearch6/pom.xml (hunk 2 above)
Comment (yanghua) [reply flink-connectors/flink-sql-connector-elasticsearch6/pom.xml]: In my memory, I used to try to remove these dependencies in the PR #9720 , but this behavior caused test case failure, so I gave up, let me try again.

**Comment by yanghua** [comment]
@aljoscha Thanks for reviewing. @twalthr and Jark have different opinions about the purpose of this PR want to do. Can you have a look at our discussion in the JIRA.

**Comment by aljoscha** [comment]
> > I don't think this will still work because this was not changed:
> > https://github.com/apache/flink/blob/10f9f1d431dbf047ea14130d9fb7fdd4f78178fc/flink-connectors/flink-connector-elasticsearch6/src/main/resources/META-INF/services/org.apache.flink.table.factories.TableFactory#L16
> > 
> > . And you changed the name of the package.
> 
> I have moved the file from `flink-connector-elasticsearch6` to `flink-sql-connector-elasticsearch6` just like what I did in the PR #9720 . So it should not be a problem?

Yes, but for the other PR you changed the service file: https://github.com/apache/flink/blob/e2c4213a8f0664285658f25f2169487eb952909e/flink-connectors/flink-sql-connector-elasticsearch7/src/main/resources/META-INF/services/org.apache.flink.table.factories.TableFactory#L16

**Comment by yanghua** [comment]
> > > I don't think this will still work because this was not changed:
> > > https://github.com/apache/flink/blob/10f9f1d431dbf047ea14130d9fb7fdd4f78178fc/flink-connectors/flink-connector-elasticsearch6/src/main/resources/META-INF/services/org.apache.flink.table.factories.TableFactory#L16
> > > 
> > > . And you changed the name of the package.
> > 
> > 
> > I have moved the file from `flink-connector-elasticsearch6` to `flink-sql-connector-elasticsearch6` just like what I did in the PR #9720 . So it should not be a problem?
> 
> Yes, but for the other PR you changed the service file:
> 
> https://github.com/apache/flink/blob/e2c4213a8f0664285658f25f2169487eb952909e/flink-connectors/flink-sql-connector-elasticsearch7/src/main/resources/META-INF/services/org.apache.flink.table.factories.TableFactory#L16


> > Yes, but for the other PR you changed the service file:

In this PR, I also changed the service file, now it is:

```
org.apache.flink.streaming.sql.connectors.elasticsearch6.Elasticsearch6UpsertTableSinkFactory
```

File: flink-connectors/flink-sql-connector-elasticsearch6/src/main/resources/META-INF/services/org.apache.flink.table.factories.TableFactory [hunk 3]
```diff
@@ -13,4 +13,4 @@
 # See the License for the specific language governing permissions and
 # limitations under the License.
 
-org.apache.flink.streaming.connectors.elasticsearch6.Elasticsearch6UpsertTableSinkFactory
+org.apache.flink.streaming.sql.connectors.elasticsearch6.Elasticsearch6UpsertTableSinkFactory
```
Comment (yanghua) [inline flink-connectors/flink-sql-connector-elasticsearch6/src/main/resources/META-INF/services/org.apache.flink.table.factories.TableFactory]: @aljoscha see here, is anything wrong?

**Comment by yanghua** [comment]
@flinkbot run travis

**Comment by yanghua** [comment]
This PR go to wrong direction, closing it.
